File: cameraview/src/main/java/com/base/cameraview/video/encoding/OutputBufferPool.java
package com.base.cameraview.video.encoding;

import android.media.MediaCodec;
import android.os.Build;

import androidx.annotation.RequiresApi;

import com.base.cameraview.internal.Pool;

/**
 * A simple {@link Pool(int, Factory)} implementation for output buffers.
 */
@RequiresApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
class OutputBufferPool extends Pool<OutputBuffer> {

    OutputBufferPool(final int trackIndex) {
        super(Integer.MAX_VALUE, new Factory<OutputBuffer>() {
            @Override
            public OutputBuffer create() {
                OutputBuffer buffer = new OutputBuffer();
                buffer.trackIndex = trackIndex;
                buffer.info = new MediaCodec.BufferInfo();
                return buffer;
            }
        });
    }
}

File: cameraview/src/main/java/com/base/cameraview/video/encoding/OutputBufferPool.kt
package com.base.cameraview.video.encoding;

import android.media.MediaCodec;

import com.base.cameraview.internal.Pool;

/**
 * A simple {@link Pool(int, Factory)} implementation for output buffers.
 */
class OutputBufferPool extends Pool<OutputBuffer> {

    OutputBufferPool(final int trackIndex) {
        super(Integer.MAX_VALUE, new Factory<OutputBuffer>() {
            @Override
            public OutputBuffer create() {
                OutputBuffer buffer = new OutputBuffer();
                buffer.trackIndex = trackIndex;
                buffer.info = new MediaCodec.BufferInfo();
                return buffer;
            }
        });
    }
}
